src/shogun/classifier/Perceptron.cpp

```diff
@@ -62,12 +63,9 @@ void Perceptron::iteration()
 	SGVector<float64_t> w = get_w();
 
 	auto labels = binary_labels(m_labels)->get_int_labels();
-	auto iter_train_labels = labels.begin();
 
-	for (const auto& v : DotIterator(features))
+	for (const auto& [v, true_label] : zip_iterator(DotIterator(features), labels))
```

Comment:
@karlnapf we can then replace all of these loops

Reply:
yes that is nice!